source/server/listener_manager_impl.cc

```diff
@@ -272,7 +272,14 @@ void ListenerImpl::setSocket(const Network::SocketSharedPtr& socket) {
   // Server config validation sets nullptr sockets.
   if (socket_ && listen_socket_options_) {
     bool ok = listen_socket_options_->setOptions(*socket_);
-    ENVOY_LOG(debug, "{}: Setting socket options {}", name_, ok ? "succeeded" : "failed");
+    const std::string message =
+        fmt::format("{}: Setting socket options {}", name_, ok ? "succeeded" : "failed");
+    if (!ok) {
```

Comment:
Do you have test coverage for both of these cases?

Reply:
Maybe not, I'll make sure I do. Sorry to make you ask for coverage again...